testing/scripts/seldon_e2e_utils.py

```diff
@@ -60,18 +68,28 @@ def get_deployment_names(sdep_name, namespace, attempts=20, sleep=5):
     # The `deploymentStatus` is dictionary which keys are names of deployments
     deployment_names = list(data["status"]["deploymentStatus"])
     logging.warning(
-        f"For SeldonDeployment {sdep_name} found following deployments: {deployment_names}"
+        f"For SeldonDeployment {sdep_name} "
+        f"found following deployments: {deployment_names}"
     )
     return deployment_names
 
 
 def wait_for_rollout(
     sdep_name, namespace, attempts=20, sleep=5, expected_deployments=1
 ):
-    deployment_names = get_deployment_names(sdep_name, namespace)
-    assert (
-        len(deployment_names) == expected_deployments
-    ), f"Expected {expected_deployments} deployment(s) but got {len(deployment_names)}"
+    deployment_names = []
+    for _ in range(attempts):
+        deployment_names = get_deployment_names(sdep_name, namespace)
+        deployments = len(deployment_names)
+
+        if deployments == expected_deployments:
+            break
```

Comment:
I guess adding `else: time.sleep(sleep)` here is necessary. Otherwise it will go through all `attempts` very fast.

Reply:
That's a good point!